### DISCUSSION TITLE
[Solved] Protection against detecting the "true" operating system?

### DISCUSSION THREAD

**Discussion by curiosityseeker** [tarampampam/random-user-agent#188] [Ideas] [answered]
Thank you for your useful add-on!

I've noticed that on some test sites the "true" operating system is detected. For example on:

https://www.deviceinfo.me/

which reports my faked OS as "Mac OS X version 11.3.6 (32-bit)" but detects my correct OS as "Linux x86_64".

Similarly on 

https://browserleaks.com/javascript

where the platform navigator object is reported as "MacIntel" but the oscpu is again recognized as "Linux x86_64".

I wonder if something can be done against this?

**Comment by tarampampam** [comment, marked as answer]
Hi @curiosityseeker!

Could you please [make an issue](https://github.com/tarampampam/random-user-agent/issues/new/choose) with reproducing steps? Currently, I can't reproduce the described behavior (Chrome 95.0.4638.54)

![image](https://user-images.githubusercontent.com/7326800/141284218-d3d80a65-2907-43e4-be99-af3c8711fd86.png)

**Reply by tarampampam** [reply]
The [fixed release (v3.3.1)](https://github.com/tarampampam/random-user-agent/releases/tag/v3.3.1) has been published

**Reply by tarampampam** [reply]
@curiosityseeker Please, mark your question as a "solved" if all is ok after the update

**Comment by curiosityseeker** [comment]
Thanks again for the quick fix!
